acceptance/api/v1/services_test.go

```diff
@@ -260,4 +264,126 @@ var _ = Describe("Services API Application Endpoints", func() {
+			Expect(service["put_key1"]).To(Equal("put_value"))
+			Expect(service).ToNot(HaveKey("username"))
+		})
+
+		It("returns a 404 when the namespace does not exist", func() {
+			response, err := env.Curl("PUT", fmt.Sprintf("%s%s/namespaces/idontexist/services/%s",
+				serverURL, api.Root, svc1), strings.NewReader(changeRequest))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+		})
+
+		It("returns a 404 when the service does not exist", func() {
+			response, err := env.Curl("PUT", fmt.Sprintf("%s%s/namespaces/%s/services/bogus",
+				serverURL, api.Root, namespace), strings.NewReader(changeRequest))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+		})
+
+		When("service is bound to an app", func() {
+			var app1 string
+			BeforeEach(func() {
+				app1 = catalog.NewAppName()
+
+				env.MakeContainerImageApp(app1, 1, containerImageURL)
+				env.BindAppService(app1, svc1, namespace)
+			})
+
+			AfterEach(func() {
+				env.DeleteApp(app1)
+			})
+
+			Describe("workload restarts", func() {
+				It("only restarts the app if the service has changed", func() {
+					getPodName := func(namespace, app string) (string, error) {
+						return helpers.Kubectl("get", "pods", "-n", namespace, "-l", fmt.Sprintf("app.kubernetes.io/name=%s", app), "-o", "jsonpath='{.items[*].metadata.name}'")
+					}
+
+					oldPodName, err := getPodName(namespace, app1)
+					Expect(err).ToNot(HaveOccurred())
+
+					response, err := env.Curl("PUT", fmt.Sprintf("%s%s/namespaces/%s/services/%s",
+						serverURL, api.Root, namespace, svc1), strings.NewReader(changeRequest))
+					Expect(err).ToNot(HaveOccurred())
+					Expect(response).ToNot(BeNil())
+					Expect(response.StatusCode).To(Equal(http.StatusOK))
+
+					var newPodName string
+
+					Eventually(func() string {
+						newPodName, err := getPodName(namespace, app1)
+						Expect(err).ToNot(HaveOccurred())
+						return newPodName
+					}).ShouldNot(Equal(oldPodName))
+
+					// Now try with no changes
+					oldPodName, err = getPodName(namespace, app1)
+					Expect(err).ToNot(HaveOccurred())
+
+					response, err = env.Curl("PUT", fmt.Sprintf("%s%s/namespaces/%s/services/%s",
+						serverURL, api.Root, namespace, svc1), strings.NewReader(changeRequest))
+					Expect(err).ToNot(HaveOccurred())
+					Expect(response).ToNot(BeNil())
+					Expect(response.StatusCode).To(Equal(http.StatusOK))
+
+					newPodName, err = getPodName(namespace, app1)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(newPodName).To(Equal(oldPodName))
```

Comment:
Since the previous check required an `Eventually`, I think this check needs a `Consistently` to be fair.

Reply:
> > @thardeck For this use case PUT on existing resources works fine for the UI. I think consistency is also important here, but I think no other PUT supports it too?
> 
> We do not have any other PUT besides this one. But I would not mind allowing creation. I will raise it in the daily.

@richard-cox The consensus was to not allow creation of a non existing service with `PUT` because we already have an endpoint for this and want to keep it consistent.